providers/tests/amazon/aws/transfers/test_http_to_s3.py

```diff
@@ -60,12 +60,34 @@ def test_execute(self, requests_mock):
         conn = boto3.client("s3")
         conn.create_bucket(Bucket=self.s3_bucket)
         operator = HttpToS3Operator(
-            task_id="s3_to_file_sensor",
+            task_id="http_to_s3_operator",
+            http_conn_id=self.http_conn_id,
+            endpoint=self.endpoint,
+            s3_key=self.s3_key,
+            s3_bucket=self.s3_bucket,
+            dag=self.dag,
+        )
+        operator.execute(None)
+
+        objects_in_bucket = conn.list_objects(Bucket=self.s3_bucket, Prefix=self.s3_key)
+        # there should be object found, and there should only be one object found
+        assert len(objects_in_bucket["Contents"]) == 1
+        # the object found should be consistent with dest_key specified earlier
+        assert objects_in_bucket["Contents"][0]["Key"] == self.s3_key
+
+    @mock_aws
+    def test_execute_stream(self, requests_mock):
+        requests_mock.register_uri("GET", EXAMPLE_URL, content=self.response)
+        conn = boto3.client("s3")
+        conn.create_bucket(Bucket=self.s3_bucket)
+        operator = HttpToS3Operator(
+            task_id="http_to_s3_operator",
             http_conn_id=self.http_conn_id,
             endpoint=self.endpoint,
             s3_key=self.s3_key,
             s3_bucket=self.s3_bucket,
             dag=self.dag,
+            extra_options={"stream": True},
         )
         operator.execute(None)
 
```

Comment:
Should you add an assert to ensure that the s3 `load_bytes` was called once instead of `load_file_obj`?

Reply:
@o-nikolas Just to clarify - we want to explicitly check that different boto3 methods were called even if we do not mock them out?

Technically should be doable.

In practical terms I'd expect that coverage data from test suite would be enough.

Please confirm if you see this as a blocker or if this is just an open question.

Reply:
should we merge @o-nikolas ?

Reply:
@potiuk still waiting on the requested changes from @rogalski

Reply:
This will be delayed, I am struggling to get local dev env operational.

Reply:
What's the problem @rogalski ? Can we help somehow? For the local dev env now we recommend `uv` and simple `uv sync --extra amazon` should **just** work -> it shoudl autoamatically install `python`, create `.venv` with all necessary libraries for amazon. See our "local virtualenv" documentation in contributing guide: https://github.com/apache/airflow/blob/main/contributing-docs/07_local_virtualenv.rst

Reply:
@potiuk solved, thanks.

All changes pushed CC @o-nikolas